### PR TITLE
fix: losing aliasMap in 'partial' subqueries

### PR DESCRIPTION
**Pull request by falkenhawk**
_ported from https://github.com/ovos/objection.js/pull/13_

This fixes an issue with 'partial' subqueries that they lose `aliasMap` from `q._context` (QueryBuilder's "internal context") while generating nested `where` sql part. `QueryBuilder.subqueryOf()` used to grab context from "parent" query, but it doesn't anymore. (since 2019-02-13) https://github.com/Vincit/objection.js/commit/524e0bef5f75a280aa54d1ee30b6b36c304ea6ad

In our case, we were getting SQL errors where alias was necessary to make it work, but it was missing
```
ER_NON_UNIQ_ERROR: Column 'event_id' in IN/ALL/ANY subquery is ambiguous
```

Basically reintroducing what was deleted in https://github.com/Vincit/objection.js/commit/524e0bef5f75a280aa54d1ee30b6b36c304ea6ad (to fix https://github.com/Vincit/objection.js/issues/1205) but wrap merging context with a condition that it should only apply to "partial" subqueries + reverse the sequence that `isPartial()` is set before `subqueryOf()`

### CODE REVIEW

File: lib/utils/objectUtils.js [hunk 1]
```diff
@@ -373,6 +373,18 @@ function isSafeKey(key) {
   return isNumber(key) || (isString(key) && key !== '__proto__');
 }
 
+function mergeMaps(map1, map2) {
+  const map = new Map(map1);
+
+  if (map2) {
+    for (let key of map2.keys()) {
```
Comment (glensc) [inline lib/utils/objectUtils.js:380]:
```suggestion
    for (const key of map2.keys()) {
```

**Comment by falkenhawk** [comment]
@lehni I've added a test which fails with
```
DBError: select `m1`.* from `model2` as `m1` where (`id_col` = (select `m2`.`id_col` from `model2` as `m2` where `m2`.`model2_prop2` = `model2`.`model2_prop2`)) order by `m1`.`model2_prop2` asc - SQLITE_ERROR: no such column: model2.model2_prop2
```
on the master branch.

Perhaps the `${query.tableRefFor(query.modelClass())}.model2_prop2` reference look weird, but it is useful e.g. for modifiers which add a where condition with subquery, referencing a column from parent query, but the alias which parent query uses is not known and cannot be hardcoded in the modifier.
